Specs/Core/S2CellSpec.js

```diff
@@ -0,0 +1,282 @@
+    expect(function () {
+      S2Cell.getLevel(BigInt("-1"));
+    }).toThrowDeveloperError();
+    expect(function () {
+      S2Cell.getLevel();
+    }).toThrowDeveloperError();
+    expect(function () {
+      S2Cell.getLevel(BigInt("3170534137668829184444"));
+    }).toThrowDeveloperError();
+    expect(function () {
+      S2Cell.getLevel(BigInt(0));
+    }).toThrowDeveloperError();
+  });
+
+  it("gets correct parent of cell", function () {
+    var cell = new S2Cell(BigInt("3383782026967515136"));
+    var parent = cell.getParent();
+    expect(parent._cellId).toEqual(BigInt("3383782026971709440"));
+  });
+
+  it("gets correct parent of cell at given level", function () {
+    var cell = new S2Cell(BigInt("3383782026967056384"));
+    var parent = cell.getParentAtLevel(21);
+    expect(parent._cellId).toEqual(BigInt("3383782026967252992"));
+    parent = cell.getParentAtLevel(7);
+    expect(parent._cellId).toEqual(BigInt("3383821801271328768"));
+    parent = cell.getParentAtLevel(0);
+    expect(parent._cellId).toEqual(BigInt("3458764513820540928"));
+  });
+
+  it("throws on getting parent of cell at invalid level", function () {
+    var cell = new S2Cell(BigInt("3458764513820540928"));
+    expect(function () {
+      cell.getParentAtLevel(0);
+    }).toThrowDeveloperError();
+
+    cell = new S2Cell(BigInt("3383782026967072768"));
+    expect(function () {
+      cell.getParentAtLevel(-1);
+    }).toThrowDeveloperError();
+
+    expect(function () {
+      cell.getParentAtLevel(30);
+    }).toThrowDeveloperError();
+  });
+
+  it("throws on getting parent of level 0 cells", function () {
+    var cell = S2Cell.fromToken("3");
+    expect(function () {
+      cell.getParent();
+    }).toThrowDeveloperError();
+  });
+
+  it("gets correct children of cell", function () {
+    var cell = new S2Cell(BigInt("3383782026971709440"));
+    var expectedChildCellIds = [
+      BigInt(3383782026959126528),
+      BigInt(3383782026967515136),
+      BigInt(3383782026975903744),
+      BigInt(3383782026984292352),
+    ];
+    var i;
+    for (i = 0; i < 4; i++) {
+      expect(cell.getChild(i)._cellId).toEqual(expectedChildCellIds[i]);
+    }
+  });
+
+  it("throws on invalid child index in getting children of cell", function () {
+    var cell = new S2Cell(BigInt("3383782026971709440"));
+    expect(function () {
+      cell.getChild(4);
+    }).toThrowDeveloperError();
+    expect(function () {
+      cell.getChild(-1);
+    }).toThrowDeveloperError();
+  });
+
+  it("throws on getting children of level 30 cell", function () {
+    var cell = new S2Cell(BigInt("3383782026967071427"));
+    expect(cell._level).toEqual(30);
+    expect(function () {
+      cell.getChild(0);
+    }).toThrowDeveloperError();
+  });
+
+  it("gets correct center of cell", function () {
+    expect(S2Cell.fromToken("1").getCenter()).toEqual(
+      Cartesian3.fromDegrees(0, 0)
+    );
```

Comment:
should these be `toEqualEpsilon` too? even though the cartographic degrees may be an integer, not sure what operations are involved in cartographic -> cartesian, let alone `getCenter()`